### ISSUE TITLE
Use type signatures everywhere

### ISSUE THREAD

**Issue by eeue56** [elmbridge/curriculum#72]
e.g from
```
update msg model =
    case msg of
        SetCurrentText newText ->
            -- currently, this does nothing!
            model
```
on https://github.com/elmbridge/curriculum/blob/master/Making%20It%20Dynamic.md

Using type signatures everywhere keeps the material consistent, as well as helping them seem less scary.

**Comment by raorao** [comment]
:+1:. if we go signatures everywhere, it's probably worth having a diagram early on in the curriculum that tells students that the signatures are helpful, but not required. 

for now, tagging this as an enhancement

**Comment by tesk9** [comment]
Possibly worth noting:

Right now, about a third of the intro-to-elm-part-1 stuff has type signatures, and they kind of just appear starting in the middle.

**Comment by avh4** [comment]
(I added type signature to all the intro-to-elm-part-1 stuff except the records section)

> also, FYI, the reason I didnt put types on the Records section in the morning exercise was that the nested record type was huge, and putting the type for the first one seemed to make the question too trivial.    Also see https://github.com/elmbridge/curriculum/issues/32 (I think it’ll probably be better to replace the existing record exercsies with some that use a `Person` record type alias for all (or at least most) of the questions; and if we did that, having type annotations in that section would be much nicer

**Comment by eeue56** [comment]
> putting the type for the first one seemed to make the question too trivial

Maybe get rid of the question, then make another question be like "write the type alias for this record", once types aliases have been introduced?

**Comment by avh4** [comment]
Yeah, https://github.com/elmbridge/curriculum/issues/32 was some ideas for replacing the current record questions.
